src/Microsoft.TestPlatform.CommunicationUtilities/SocketCommunicationManager.cs

```diff
@@ -138,7 +138,11 @@ public async Task AcceptClientAsync()
         /// <returns>True if Client is connected, false otherwise</returns>
         public bool WaitForClientConnection(int clientConnectionTimeout)
         {
-            return this.clientConnectedEvent.WaitOne(clientConnectionTimeout);
+            Stopwatch stopWatch = Stopwatch.StartNew();
```

Comment:
It seems weird to me that we start a timer and never really pause/stop it. Is it expected?

Reply:
I think that the idea is not to start a timer but measure the time, and it's done with stopWatch.ElapsedMilliseconds(implicit stop/read)

Reply:
I will keep it as is.

@Evangelink  Technically you are right, this is adding few ticks between the time the WaitOne returns, and we capture the value. But in reality it does not matter, we are looking at the number in milliseconds and don't really care if 1.00004 ms or 1.00013 ms is the correct measurement. So this just saves one line of code. Stopwatch also has no overhead when we don't stop it, because it is not doing anything until you ask it for time, so doing it like this is fine.

Reply:
The time diff wasn't the main concern.

> Stopwatch also has no overhead when we don't stop it, because it is not doing anything until you ask it for time, so doing it like this is fine.

This was, and I thought were would be some overhead of having timers running in the background for some time but as you say the overhead is neglectable, fine to keep as-is.

Reply:
Just to be clear. There are no timers inside of stopwatch. It asks for the current timestamp when started, and then asks for another when you call ElapsedTime or Stop. It reports the difference between those two. 

Apart from the call to get timestamp, and having one more object created, there is no permanent overhead of stopwatch running in background.